File: Back-End/src/main/java/com/bloom/demo/service/hospital/FeeService.java
package com.bloom.demo.service.hospital;

import com.bloom.demo.model.StatNumbers;
import com.bloom.demo.model.hospital.Fee;
import com.bloom.demo.model.hospital.FeePatient;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface FeeService {

    Mono<Fee> createFee(Fee fee);

    Mono<Fee> getFeeByNumber(String feeNumber);

    Mono<Fee> updateFee(Fee fee);

    Mono<Void> deleteFeeByNumber(String feeNumber);

    Flux<Fee> getFeeByPatientId(String patientId);

    Mono<Double> getPatientFinalPrice(String patientId);

    Mono<Double> getAllFees();

    Flux<FeePatient> getAll();

    Mono<Double> getTotalBeforeToday();

    Mono<StatNumbers> getFeeAnalysis();
}

File: Front-End/src/app/fee.service.ts
import {Injectable} from '@angular/core';
import {HttpClient, HttpResponse} from '@angular/common/http';
import {Observable} from 'rxjs';
import {Fee} from './model/Fee';

@Injectable({
  providedIn: 'root'
})
export class FeeService {

  private feeURL = 'http://localhost:8080/hospital/fee/';

  constructor(private http: HttpClient) {
  }

  getFeeByNumber = (feeNumber: number): Observable<Fee> => this.http.get<Fee>(this.feeURL + feeNumber);

  createFee = (fee: Fee): Observable<Fee> => this.http.post<Fee>(this.feeURL, fee);

  updateFee = (fee: Fee): Observable<Fee> => this.http.put<Fee>(this.feeURL, fee);

  deleteFee = (feeNumber: number): Observable<HttpResponse<any>> => this.http.delete(this.feeURL + feeNumber, {observe: 'response'});

  getPatientFeesByPatientId = (patientId: number): Observable<Array<Fee>> =>
    this.http.get<Array<Fee>>(this.feeURL + 'patient/' + patientId);

  getPatientFeesAmountByPatientId = (patientId: number): Observable<number> =>
    this.http.get<number>(this.feeURL + 'patient/price/' + patientId);

  getAll = (): Observable<Array<FeePatient>> => this.http.get<Array<FeePatient>>(this.feeURL);

  getFeeTotalInDay = (): Observable<number> => this.http.get<number>(this.feeURL + 'yesterday');

  getFeeAnalysis = (): Observable<StatNumbers> => this.http.get<StatNumbers>(this.feeURL + 'analysis');
}
